packages/react-email/src/cli/utils/preview/hot-reloading/create-dependency-graph.ts

```diff
@@ -133,9 +143,12 @@ export const createDependencyGraph = async (directory: string) => {
             const pathWithExtension = checkFileExtensionsUntilItExists(
               pathToDependencyFromDirectory,
             );
+
             if (pathWithExtension) {
               pathToDependencyFromDirectory = pathWithExtension;
-            } else if (isDev) {
```

Comment:
this `else` here is required because the other if statement doesn't have any early return